sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -393,7 +393,8 @@ case class SortMergeJoinExec(
       input: Seq[Attribute]): Seq[ExprCode] = {
     ctx.INPUT_ROW = row
     ctx.currentVars = null
-    keys.map(BindReferences.bindReference(_, input).genCode(ctx))
+    val inputAttributeSeq: AttributeSeq = input
+    keys.map(BindReferences.bindReference(_, inputAttributeSeq).genCode(ctx))
```

Comment:
why not `toBoundExprs(keys, input).map(_.genCode(ctx))`

Reply:
It should be...